docs/providers/aws/guide/variables.md

```diff
@@ -274,6 +274,37 @@ custom:
 
 In this example, the serverless variable will contain the decrypted value of the secret.
 
+Variables can also be object, since AWS Secrets Manager can store secrets not only in plain text but also in JSON.
+
+If the above secret `secret_ID_in_Secrets_Manager` is something like below,
+
+```json
+{
+  "num": 1,
+  "str": "secret",
+  "arr": [true, false]
+}
+```
+
+variables will be resolved like
+
+```yml
+service: new-service
+provider: aws
+functions:
+  hello:
+    name: hello
+    handler: handler.hello
+custom:
+  supersecret: 
```

Comment:
How hard would it be to add `${ssm:/aws/reference/secretsmanager/secret_ID_in_Secrets_Manager~true:hello}`

Is the current work around to put it in custom then elsewhere refrence `${self:custom.hello}`?

Reply:
I think it would involve some "caching" so it's not requesting SSM every time this line is references. Using `custom` seems more elegant!